### PR TITLE
Delete old non approved Functions when approved version is saved.

### LINKED ISSUE #195: Delete Function drafts when saving as public
Drafts of a `Function` model instance should be deleted permanently instead of soft deletion. Currently delete request to the Function API endpoint changes the instance state to `deleted` effectively soft deleting the instance. Replace this soft delete logic with real deletion.

### PR DESCRIPTION
**Pull request by ehaivala**
To reduce bloat on database level, remove old non approved `Function`s when new approved version is saved. Old `deleted` state is also replaced by this. Instances with `deleted` state will be deleted when applying migrations. Add management command to delete all non approved `Function`s that are older than the newest approved `Function`.

Resolves #195

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ local_settings.py
 htmlcov
 .cache
 .DS_Store
+.pytest_cache/
```
Comment (japauliina) [inline .gitignore]: Newline missing.

File: metarecord/models/function.py [hunk 2]
```diff
@@ -18,20 +19,31 @@ def filter_for_user(self, user):
...
+            version__lt=function.version,
+            classification=function.classification
+        )
+
+    def non_approved(self):
+        return self.filter(
+            Q(state=Function.DRAFT) |
+            Q(state=Function.SENT_FOR_REVIEW) |
+            Q(state=Function.WAITING_FOR_APPROVAL)
+        )
```
[the remark this answers is not in the document]
Comment (ehaivala) [reply metarecord/models/function.py]: I don't know what I was thinking when I came up with this. 😄

File: metarecord/migrations/0039_remove_deleted_state.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('metarecord', '0038_disable_unique_attribute_value_index'),
+    ]
+
+    operations = [
+        migrations.RunPython(delete_soft_deleted_functions, no_op),
```
[the remark this answers is not in the document]
Comment (ehaivala) [reply metarecord/migrations/0039_remove_deleted_state.py]: Cool. I'll use that instead.

**Comment by ehaivala** [comment]
I'll check if that change history deletion is indeed something we want as well. I think it is wanted but better to be safe than sorry.

**Comment by ehaivala** [comment]
Got confirmation that this is okay.

> Got an answer from tiedonhallintapäällikkö [Chief of Information Management], we can delete draft-versions and version history regarding drafts when approved version is saved.

**Pull request merged**